src/layout/components/NavHeader/Language.vue

```diff
@@ -69,12 +69,16 @@ export default {
     changeMomentLang() {
       if (this.currentLang.code.indexOf('en') > -1) {
         this.$moment.locale('en')
+        document.documentElement.lang = 'en'
       } else if (this.currentLang.code.indexOf('ja') > -1) {
         this.$moment.locale('ja')
+        document.documentElement.lang = 'ja'
       } else if (this.currentLang.code.indexOf('zh_hant') > -1) {
         this.$moment.locale('zh-tw')
+        document.documentElement.lang = 'zh-tw'
       } else {
         this.$moment.locale('zh-cn')
+        document.documentElement.lang = 'zh-cn'
       }
     },
     changeLangTo(item) {
```

Comment:
No, there is no indication of any regularities, potential issues or optimization suggestions in the code provided. The code snippet seems to be correctly implementing changes to moment.js locale functionality based on the current language setting stored in $locale. However, it might benefit from being more concise since its structure is straightforward with only two functions.

In terms of improvements, here's an optimized suggestion which uses arrow function for better readability:

```javascript
changeMomentLang = () => ({$moment, document}).invoke(['useLocale', 'setLanguage'], ['en', 'ja'][this.currentLang.isEnglish ? 0 : this.currentLang.isJapanese ? 1 : 2])
```

However, keep in mind that without access to actual execution times or performance metrics of your app, it would not be accurate to say whether some aspect of the earlier version was faster/less resource-intensive than another.